### PR TITLE
EVG-6829: Add dependencies check prior to dispatching a task via the Revised Dispatcher

### PR DESCRIPTION
**Pull request by StuartButler**

### CODE REVIEW

File: model/task_queue_service.go [hunk 1]
```diff
@@ -308,13 +328,53 @@ func (t *basicCachedDispatcherImpl) FindNextTask(spec TaskSpec) *TaskQueueItem {
...
 
 				return nil
 			}
 
 			// A non-task group schedulableUnit's tasks ([]TaskQueueItem) only contain a single element.
+			item := unit.tasks[0]
+
+			nextTaskFromDB, err := task.FindOneId(item.Id)
+			if err != nil {
+				grip.Error(message.WrapError(err, message.Fields{
```
Comment (tychoish) [inline model/task_queue_service.go]: shouldn't be an error. 

can we just have a function in the task finder that returns a bool,err (for "is this ID met?")

File: model/task_queue_service.go [hunk 2]
```diff
@@ -308,13 +328,53 @@ func (t *basicCachedDispatcherImpl) FindNextTask(spec TaskSpec) *TaskQueueItem {
...
+					"dispatcher": SchedulableUnitDispatcher,
+					"function":   "FindNextTask",
+					"message":    "problem finding task in db",
+					"task_id":    item.Id,
+					"distro_id":  d.distroID,
+				}))
+				return nil
+			}
+			if nextTaskFromDB == nil {
+				grip.Error(message.Fields{
```
Comment (tychoish) [inline model/task_queue_service.go]: reduce logging level

File: model/task_queue_service.go [hunk 3]
```diff
@@ -376,27 +436,53 @@ func (t *basicCachedDispatcherImpl) nextTaskGroupTask(unit schedulableUnit) *Tas
 		nextTaskFromDB, err := task.FindOneId(nextTaskQueueItem.Id)
 		if err != nil {
 			grip.Error(message.WrapError(err, message.Fields{
```
Comment (tychoish) [inline model/task_queue_service.go]: reduce these logging levels.

File: model/task_queue_service.go (hunk 1 above)
Comment (StuartButler) [reply model/task_queue_service.go]: Standardising to `Error.Warning`

File: model/task_queue_service.go (hunk 2 above)
Comment (StuartButler) [reply model/task_queue_service.go]: Standardising to `Error.Warning`

File: model/task_queue_service.go (hunk 3 above)
Comment (StuartButler) [reply model/task_queue_service.go]: Standardising to `Error.Warning`

File: model/task_queue_service.go (hunk 1 above)
Comment (StuartButler) [reply model/task_queue_service.go]: Leaving as is per 1:1 discussion.

**Comment by tychoish** [comment]
This seems fine, can there be a ticket number?

**Comment by StuartButler** [comment]
Oops - the ticket number is back in the PR title once more.

**Pull request merged**